pkg/testing/fetcher_artifact.go

```diff
@@ -102,56 +109,117 @@ func (r *artifactResult) Name() string {
 	if err != nil {
 		return fmt.Errorf("failed to download %s: %w", r.src, err)
 	}
 
 	// fetch package asc
-	err = DownloadPackage(ctx, l, r.doer, r.src+extAsc, filepath.Join(dir, r.path+extAsc))
+	err = DownloadPackage(ctx, l, r.doer, r.src+extAsc, dst+extAsc)
 	if err != nil {
 		return fmt.Errorf("failed to download %s: %w", r.src, err)
 	}
 
 	return nil
 }
 
-func findURI(ctx context.Context, doer httpDoer, version string) (string, error) {
-	version, buildID := splitBuildID(version)
-	artifactsURI := fmt.Sprintf("https://artifacts-api.elastic.co/v1/search/%s/elastic-agent", version)
-	req, err := http.NewRequestWithContext(ctx, "GET", artifactsURI, nil)
+type projectResponse struct {
+	Packages map[string]interface{} `json:"packages"`
+}
+
+type projectsResponse struct {
+	ElasticPackage projectResponse `json:"elastic-agent-package"`
+}
+
+type manifestResponse struct {
+	Projects projectsResponse `json:"projects"`
+}
+
+func findBuild(ctx context.Context, doer httpDoer, version *semver.ParsedSemVer) (*projectResponse, error) {
+	// e.g. https://snapshots.elastic.co/8.13.0-l5snflwr/manifest-8.13.0-SNAPSHOT.json
+	manifestURI := fmt.Sprintf("https://snapshots.elastic.co/%s-%s/manifest-%s-SNAPSHOT.json", version.CoreVersion(), version.BuildMetadata(), version.CoreVersion())
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, manifestURI, nil)
+	if err != nil {
+		return nil, err
+	}
+	resp, err := doer.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("%s; bad status: %s", manifestURI, resp.Status)
+	}
+
+	var body manifestResponse
+
+	dec := json.NewDecoder(resp.Body)
+	if err := dec.Decode(&body); err != nil {
+		return nil, err
+	}
+
+	return &body.Projects.ElasticPackage, nil
+}
+
+func findVersion(ctx context.Context, doer httpDoer, version *semver.ParsedSemVer) (*projectResponse, error) {
+	artifactsURI := fmt.Sprintf("https://artifacts-api.elastic.co/v1/search/%s/elastic-agent", version.VersionWithPrerelease())
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, artifactsURI, nil)
 	if err != nil {
-		return "", err
+		return nil, err
 	}
 	resp, err := doer.Do(req)
 	if err != nil {
-		return "", err
+		return nil, err
 	}
 	defer resp.Body.Close()
 	if resp.StatusCode != http.StatusOK {
-		return "", fmt.Errorf("%s; bad status: %s", artifactsURI, resp.Status)
+		return nil, fmt.Errorf("%s; bad status: %s", artifactsURI, resp.Status)
 	}
 
-	body := struct {
-		Packages map[string]interface{} `json:"packages"`
-	}{}
+	var body projectResponse
 
 	dec := json.NewDecoder(resp.Body)
 	if err := dec.Decode(&body); err != nil {
-		return "", err
+		return nil, err
 	}
 
-	if len(body.Packages) == 0 {
+	return &body, nil
+}
+
+func findURI(ctx context.Context, doer httpDoer, version *semver.ParsedSemVer) (string, error) {
+	var (
+		project *projectResponse
+		err     error
+	)
```

Comment:
[Suggestion]
This form isn't really used in our code, I'd suggest keeping it consistent with the rest of the code by.